src/v/features/feature_table.cc

```diff
@@ -686,10 +715,35 @@ void feature_table::set_license(security::license license) {
     _license = std::move(license);
 }
 
-void feature_table::revoke_license() { _license = std::nullopt; }
+void feature_table::set_builtin_trial_license(
+  model::timestamp cluster_creation_timestamp) {
+    if (ss::this_shard_id() == 0) {
+        vlog(featureslog.debug, "Initializing builtin trial license.");
+    }
+    _builtin_trial_license_initialized = true;
+    _builtin_trial_license = make_builtin_trial_license(
+      model::to_time_point(cluster_creation_timestamp));
```

Comment:
nit: while `make_builtin_trial_license` doesn't seem to be able to throw, we should probably set the bool after it is initialized, mainly for sanity reasons.

Reply:
Fixed